mapstruct/src/test/java/com/baeldung/expression/mapper/LicenseMapperUnitTest.java

```diff
@@ -0,0 +1,97 @@
+            .satisfies(l -> {
+                assertThat(l.getStartDate()
+                    .toLocalDate()).isEqualTo(LocalDate.now());
+                assertThat(l.getEndDate()
+                    .toLocalDate()).isEqualTo(LocalDate.now()
+                    .plusYears(1));
+            });
+    }
+
+    @Test
+    void givenLicenseDtoWithEndDateAndWithoutStartDate_WhenMapperMethodIsInvoked_ThenLicenseShouldBePopulatedWithDefaultStartDate() {
+        License license = licenseMapper.toLicense(LicenseDto.builder()
+            .endDate(LocalDateTime.now()
+                .plusYears(2))
+            .build());
+        assertThat(license).isNotNull()
+            .satisfies(l -> {
+                assertThat(l.getStartDate()
+                    .toLocalDate()).isEqualTo(LocalDate.now());
+                assertThat(l.getEndDate()
+                    .toLocalDate()).isEqualTo(LocalDate.now()
+                    .plusYears(2));
+            });
+    }
+
+    @Test
+    void givenLicenseDtoWithoutEndDateAndWithoutStartDate_WhenMapperMethodIsInvoked_ThenLicenseShouldBePopulatedWithDefaultStartDateAndEndDate() {
+        License license = licenseMapper.toLicense(LicenseDto.builder()
+            .build());
+        assertThat(license).isNotNull()
+            .satisfies(l -> {
+                assertThat(l.getStartDate()
+                    .toLocalDate()).isEqualTo(LocalDate.now());
+                assertThat(l.getEndDate()
+                    .toLocalDate()).isEqualTo(LocalDate.now()
+                    .plusYears(1));
+            });
+    }
+
+    @Test
+    void givenLicenseDtoWithoutStartDateAndEndDate_WhenMapperMethodIsInvoked_ThenLicenseShouldBePopulatedWithDefaultDetails() {
+        License license = licenseMapper.toLicense(LicenseDto.builder()
+            .build());
+        assertThat(license).isNotNull()
+            .satisfies(l -> {
+                assertThat(l.getStartDate()
+                    .toLocalDate()).isEqualTo(LocalDate.now());
+                assertThat(l.getEndDate()
+                    .toLocalDate()).isEqualTo(LocalDate.now()
+                    .plusYears(1));
+                assertThat(l.isActive()).isTrue();
+                assertThat(l.isRenewalRequired()).isFalse();
+            });
+    }
+
+    @Test
+    void givenLicenseDtoWithEndDateInTwoWeeks_WhenMapperMethodIsInvoked_ThenLicenseShouldBePopulatedWithReminderSetToTrue() {
+        License license = licenseMapper.toLicense(LicenseDto.builder()
+            .endDate(LocalDateTime.now()
+                .plusDays(10))
+            .build());
+        assertThat(license).isNotNull()
+            .satisfies(l -> {
+                assertThat(l.getStartDate()
+                    .toLocalDate()).isEqualTo(LocalDate.now());
+                assertThat(l.getEndDate()
+                    .toLocalDate()).isEqualTo(LocalDate.now()
+                    .plusDays(10));
+                assertThat(l.isActive()).isTrue();
```

Comment:
The focus of the test is isRenewalRequired, these assertions can be removed for clarity